Sources/PaymentsCore/Networking/HTTP.swift

```diff
@@ -0,0 +1,6 @@
+import UIKit
+
+protocol HTTP {
+
+    func send(_ urlRequest: URLRequest, completion: @escaping (HTTPResponse) -> Void)
+}
```

Comment:
I'd include the `API` protocol in this list as well - the only difference between `API` and `HTTP` is that one takes a `URLRequest` and the other an `APIRequest` (which itself describes most of what is required to make a URLRequest). I feel like there's some potential simplification here that would be make this a bit more straightforward

Reply:
`HTTPRequest` and `HTTPResponse` are meant to encapsulate the networking layer into our domain. Since they're plain value objects, they are good assertion vehicles when unit testing. The same goes for `APIRequest`. The only awkward part (on Android too) is there is no _APIResponse_ which breaks the symmetry, but having access to the HTTPResponse was sufficient enough.

Reply:
Yeah the protocols are specifically for injecting mocks. The naming isn't final `HTTPClientProtocol` and `APIClientProtocol` felt kind of verbose.